ir_datasets/datasets/trec_fair_2021.py

```diff
@@ -0,0 +1,179 @@
+import json
+import mwparserfromhell
+import codecs
+from typing import NamedTuple, Dict
+import ir_datasets
+from ir_datasets.util import GzipExtract, Cache, Lazy
+from ir_datasets.datasets.base import Dataset, YamlDocumentation, FilteredQueries
+from ir_datasets.formats import BaseQueries, BaseDocs, BaseQrels, TrecQrel
+from ir_datasets.indices import PickleLz4FullStore
+import pyautocorpus
+from itertools import chain
+
+
+_logger = ir_datasets.log.easy()
+
+
+NAME = 'trec_fair_2021'
+
+
+class FairTrecDoc(NamedTuple):
+    doc_id: str
+    title: str
+    text: str
+    markupfreetext: str
+    url: str
+    quality_score: str
+    geographic_locations: str
+    quality_score_disk: str
+
+
+class FairTrecQuery(NamedTuple):
+    query_id: str
+    text: str
+
+#    metadata: Dict[str, str]
+
+
+class FairTrecDocs(BaseDocs):
+    def __init__(self, name, dlc,mlc):
+        super().__init__()
+        self._name = name
+        self._dlc = dlc
+        self._mlc = mlc
+
+    def docs_iter(self):
+        return iter(self.docs_store())
+
+    def _docs_iter(self):
+        data2 = {}
+        with self._mlc.stream() as stream2:
+            
+            for line in stream2:
+                dataContx = json.loads(line) 
+                data2[dataContx["page_id"]] = dataContx
+                
+                
+                
+        
+        with self._dlc.stream() as stream1 :
+            textifier = pyautocorpus.Textifier()
+            for line1 in stream1:
+                data1 = json.loads(line1)
+               
+                
+                match = data2.get(int(data1["id"]))
+              
+                try:
+                    if match:
+                    
+                        plaintext = ""
+                        plaintext = textifier.textify(data1['text'])
+                        yield FairTrecDoc(data1['id'], data1['title'],data1['text'],plaintext, data1['url'], str(match['quality_score']), match['geographic_locations'], str(match['quality_score_disc']))
+                    else: 
+                   
+                        plaintext = ""
+                        plaintext = textifier.textify(data1['text'])
+                        yield FairTrecDoc(data1['id'], data1['title'],data1['text'],plaintext, data1['url'], "NA","NA","NA")
+                except ValueError as err:
+                    print("[{} - {}]: {}".format(data1['id'], data1['title'], err))
```

Comment:
Doesn't this mean we're missing some documents? I think we're better off handling errors we know will come up, namely the one with the comments.

```
    except ValueError as err:
        message, position = err.args
        if message == "Expected markup type 'comment'":
            # unmatched <!-- comment tag
            # The way Wikipedia renders this is it cuts the article off at this point.
            # We'll follow that here, given it's only 22 articles of the 6M.
            # (Note: the position is a byte offset, so that's why it encodes/decodes.)
            textifier.textify(data['text'].encode()[:position].decode())
        else:
            raise
```

You can update the code so the try-catch only surrounds the call to `textify`